src/views/DatasetManager/components/DatasetOverview/components/CategoryAccordion/components/CategoryDetailsDialog/CategoryDetailsDialog.js

```diff
@@ -0,0 +1,162 @@
+      previousAppError.current = applicationError;
+      clearApplicationErrorMessage();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open, clearApplicationErrorMessage]); // Do not include applicationError in deps list
+
+  const handleOpen = useCallback(() => {
+    setOpen(true);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    if (previousAppError.current != null) setApplicationErrorMessage(previousAppError.current);
+    else clearApplicationErrorMessage();
+
+    setOpen(false);
+  }, [clearApplicationErrorMessage, setApplicationErrorMessage]);
+
+  const categoryKpis = useMemo(() => {
+    return (
+      <Grid container sx={{ p: 5 }} spacing={8} justifyContent="center">
+        {(category?.kpis ?? []).map((kpiMetadata) => {
+          const kpi = kpis.find((kpiResult) => kpiResult.id === kpiMetadata.id) ?? kpiMetadata;
+          return (
+            <Grid item key={`${kpi.id}-key`}>
+              <GraphIndicator categoryId={category.id} id={kpi.id} kpi={kpi} />
+            </Grid>
+          );
+        })}
+      </Grid>
+    );
+  }, [category.id, category?.kpis, kpis]);
+
+  const detailsTable = useMemo(() => {
+    const query = getQuery(category?.previewTable?.queryId)?.query;
+    if (query == null) return null;
+
+    const parameterData = {
+      id: category?.id,
+      options: {
+        columns: category?.previewTable?.columns,
+        dynamicValues: { query, resultKey: category?.previewTable?.resultKey },
+      },
+    };
+    const tableOptions = {
+      buttons: {
+        label: false,
+        import: false,
+        addRow: false,
+        deleteRow: false,
+        revert: false,
+      },
+      height: '100%',
+    };
+
+    return (
+      <GenericTable
+        parameterData={parameterData}
+        context={{ isDarkTheme, editMode: false, targetDatasetId: datasetId, tableOptions }}
+        parameterValue={{ status: TABLE_DATA_STATUS.EMPTY }}
+        setParameterValue={() => {}}
+        resetParameterValue={() => {}}
+        isDirty={null}
+      />
+    );
+  }, [
+    category?.previewTable?.queryId,
+    getQuery,
+    isDarkTheme,
+    datasetId,
+    category?.id,
+    category?.previewTable?.columns,
+    category?.previewTable?.resultKey,
+  ]);
+
+  return (
+    detailsTable && (
+      <>
+        <Button data-cy="category-details-dialog-open-button" variant="outlined" onClick={handleOpen}>
+          {t('commoncomponents.dialog.share.dialog.buttons.open', 'Open')}
+        </Button>
+        <Dialog
+          data-cy="category-details-dialog"
+          disableEnforceFocus
+          fullScreen
+          open={open}
+          onClose={handleClose}
+          fullWidth={true}
+          maxWidth={'xl'}
+          PaperProps={{
+            elevation: 1,
+            style: {
+              minHeight: '100%',
+              maxHeight: '100%',
+            },
+          }}
+        >
+          <DialogTitle>
+            <ApplicationErrorBanner />
```

Comment:
Do we still need this component here even with all errors handled in placeholder?